vllm/worker/cache_engine.py

```diff
@@ -85,16 +89,21 @@ def allocate_cpu_cache(self) -> List[KVCache]:
         cpu_cache: List[KVCache] = []
         key_block_shape = self.get_key_block_shape()
         value_block_shape = self.get_value_block_shape()
+        pin_memory = not in_wsl()
+        if not pin_memory:
+            # Pinning memory in WSL is not supported.
+            # https://docs.nvidia.com/cuda/wsl-user-guide/index.html#known-limitations-for-linux-cuda-applications
+            logger.warn("Running in WSL, not pinning CPU cache memory.")
```

Comment:
nit: Can we provide more information like this? Also, I'd like to avoid the term "CPU cache memory", as it can be misleading to users.
```suggestion
            logger.warn("Using 'pin_memory=False' as WSL is detected. This may slow down the performance.")
```